Sources/Private/Experimental/ExperimentalAnimationLayer.swift

```diff
@@ -215,6 +215,13 @@ extension ExperimentalAnimationLayer: RootAnimationLayer {
       animation.frameTime(forProgress: (presentation() ?? self).animationProgress)
     }
     set {
+      // Currently, setting `currentFrame` causes the existing layer hierarchy
+      // to be discarded and then completely rebuilt.
+      // This isn't a problem when setting the animation up just once,
+      // but can max out the CPU (causing frame drops / lag) for very-large animations
+      // when interactively controlling the current frame with a user gesture.
+      // TODO: This should be adjusted to use the existing layer hierarchy
+      //       and control something like `layer.timeOffset`.
```

Comment:
Left a note here about performance issues when interactively updating `currentFrame`: when bound to a gesture it rebuilds the entire layer hierarchy on every frame, which becomes a perf problem for large enough animations (this doesn't happen when playing an animation noninteractively, since we construct the layer hierarchy once as then Core Animation takes over).

Should be reasonably straightforward to fix some sort of `layer.timeOffset` approach, but will leave this as-is for now

![2022-01-06 15 43 13](https://user-images.githubusercontent.com/1811727/148592202-244ea030-4a17-4163-ba20-cbe2de61fd95.gif)